tensor/stats/stats/funcs.go

```diff
@@ -4,62 +4,341 @@
 
 package stats
 
-import "math"
+import (
+	"math"
 
-// These are standard StatFunc functions that can operate on tensor.Tensor
-// or table.Table, using float64 values
+	"cogentcore.org/core/tensor"
+)
 
-// StatFunc is an statistic function that incrementally updates agg
-// aggregation value from each element in the tensor in turn.
-// Returns new agg value that will be passed into next item as agg.
-type StatFunc func(idx int, val float64, agg float64) float64
+// StatsFunc is the function signature for a stats function,
+// where the output has the same shape as the input but with
+// the outermost row dimension size of 1, and contains
+// the stat value(s) for the "cells" in higher-dimensional tensors,
+// and a single scalar value for a 1D input tensor.
+// Critically, the stat is always computed over the outer row dimension,
+// so each cell in a higher-dimensional output reflects the _row-wise_
+// stat for that cell across the different rows.  To compute a stat
+// on the [tensor.SubSpace] cells themselves, must call on a
+// [tensor.New1DViewOf] the sub space.
+// All stats functions skip over NaN's, as a missing value.
+// Stats functions cannot be computed in parallel,
+// e.g., using VectorizeThreaded or GPU, due to shared writing
+// to the same output values.  Special implementations are required
+// if that is needed.
+type StatsFunc func(in, out *tensor.Indexed)
```

Comment:
We are in `stats`; what about just `Func`?

Reply:
there might be other kinds, e.g., NormFunc, within package scope.  doesn't hurt to be specific here I think -- much clearer.